lib/linkerd2-trace-context/src/layer.rs

```diff
@@ -142,7 +152,7 @@ where
         let f = self.inner.call(request);
 
         ResponseFuture {
-            trace: span.map(|span| (span, self.sink.clone())),
+            trace: span.map(|span| (span, sink.clone())),
```

Comment:
I think the `sink.clone()` here is no longer necessary, since you clone `self.sink` here:
https://github.com/linkerd/linkerd2-proxy/pull/354/files#diff-424b22001827f29238d4de967af40d6dR117

```suggestion
            trace: span.map(|span| (span, sink)),
```